### PR TITLE
refactor(btn): deprecate aria-live on btn #1701

### LINKED ISSUE #1701: button: deprecate aria-live and aria-disabled on loading button
We are making updates to the loading button pattern. We intend to remove the aria-live and aria-disabled (but keep the aria-label=loading) state. We can do this change in stages, but will need to be backwards compatible with old markup until next major version.

### PR DESCRIPTION
**Pull request by ianmcburnie**
Fixes #1701

- [X] This PR contains CSS changes
- [X] These changes are intended to be backwards compatible

## Description

Deprecates live-region on buttons in accordance with latest MIND Pattern guidance: https://ebay.gitbook.io/mindpatterns/appendix/faq#can-a-focussed-button-be-a-live-region

This prevents sequencing/announcement issues that we are encountering in various AT. 

Instead of using ARIA disabled and live-region attributes to denote a "busy" state, we now just rely on the aria-label. This should make the CSS selectors a lot easier to reason with going forward (when we get round to cleaning things up in next major).

## Notes

* I changed all references from "Loading" to "Busy"
* In the example, I now show just a single, base button in busy state instead of primary, secondary and tertiary

## Screenshots

Before

Screen Shot 2022-03-28 at 2 50 56 PM

After

Screen Shot 2022-03-28 at 2 50 39 PM

dark mode

Screen Shot 2022-03-28 at 2 52 07 PM

## Checklist

- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue

- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Review by agliga — COMMENTED** [review]
It looks good overall. 

Two issues I see
* When the loading button is set to primary and I tab through I don't see any focus ring or indication that the button is focused. I know it's not supposed to be clickable but it is still tabbable and focusable (only primary has that issue, secondary and tertiary do not). My guess it is has to do something with focus changing the background color
* The interactive example is not working. Not sure if it's due to the spinner change or due to this pr change.

**Comment by agliga** [comment]
> It looks good overall.
> 
> Two issues I see
> 
> * When the loading button is set to primary and I tab through I don't see any focus ring or indication that the button is focused. I know it's not supposed to be clickable but it is still tabbable and focusable (only primary has that issue, secondary and tertiary do not). My guess it is has to do something with focus changing the background color
> * The interactive example is not working. Not sure if it's due to the spinner change or due to this pr change.

For the interactive example, it actually is working sometimes. I think it's a cache issue because I didn't see the spinner loaded correctly on some tries but then it started working. 
Maybe we should change that one to use the inline SVG too?

**Comment by ianmcburnie** [comment]
Good call, I will switch to using the inline SVG on the example.

I will restore the focus state (bg color) when in busy state, I agree it doesn't look good in some browsers.

An alternative is to increase the outline-offset, so that all browser look more like Safari:

Screen Shot 2022-03-29 at 8 32 22 AM

..but that might be something for a separate change/PR.

**Comment by ianmcburnie** [comment]
So, I backtracked/removed the `btn--busy` class. I don't think we need it. The outcome of this is that buttons in busy state will have their usual focus/hover states. Given that the button isn't technically disabled, this is probably fine. The user can understand the button is in a loading state from its label.

It's probably worth revisiting if we want to suppress hover/focus states when the old aria-live code is cleaned up in next major version.

Latest screenshot:

Screen Shot 2022-03-29 at 3 04 26 PM

**Review by agliga — APPROVED** [review]

**Review by ArtBlue — APPROVED** [review]
Looks great!

File: docs/_includes/button.html [hunk 1]
```diff
@@ -218,30 +218,12 @@ <h3 id="button-icon-adjacent">Adjacent Icon Button</h3>
 </button>
     {% endhighlight %}
 
-    <h3 id="button-loading">Loading State for Button</h3>
-        <p>Replace the button contents with a <a href="#progress-spinner">progress spinner</a> to create a loading button state.</p>
-        <p>The button must be marked as a labelled live-region for assistive technology and becomes <a href="#button-partially-disabled">partially disabled</a> for the duration of the loading state.</p>
+    <h3 id="button-Busy">Busy State for Button</h3>
```
[the remark this answers is not in the document]
Comment (ianmcburnie) [reply docs/_includes/button.html:221]: Oops - no! Will fix, thanks.

**Pull request merged**